app/src/lib/stores/app-store.ts

```diff
@@ -1139,6 +1141,7 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return Promise.resolve(null)
     }
 
+    this.updateStashEntryCountMetric(repository)
```

Comment:
Hey @iAmWillShepherd, me and @billygriffin just looked through this today in our sync-up and noticed that there's a key difference between this and what's being proposed for 1.7 (https://github.com/desktop/desktop/pull/7438) in that the 1.7 release will trigger `updateStashEntryCountMetric` from `_refreshRepository` whereas this is called from `_selectRepository`.

The potential problem that we see with this is for the subset of our users that have a workflow similar to my own, i.e. they work primarily in one repository (rarely switch) and they keep their desktop instances running for a long time (weeks in my case) this approach will lead to us checking and incrementing the count pretty infrequently whereas the proposed solution for 1.7 we will check once every 24 hours regardless of whether they switch or not.

The end result here could be that if the segment we're talking about isn't trivial in size (I would personally guess that it's a decent chunk of our user base) we'd see a spike in stashes created outside of desktop once we ship 1.7 which would be the opposite of what we'd want.

I would propose we amend the 1.6.6 branch with a change that moves this check to the same place as it will be in 1.7 in an effort to avoid collection inconsistencies between versions.

Reply:
> I would propose we amend the 1.6.6 branch with a change that moves this check to the same place as it will be in 1.7 in an effort to avoid collection inconsistencies between versions.

I'd rather move the check in the 1.7 PR to `selecectRepository`. For context, the PR from 1.6.6 initially had the check in `refreshRepository`, but I moved it because that code path get's ran a lot and doing DB queries isn't all that fast.

I'm going to link this comment in the PR for 1.7 to make sure I make this change.

Reply:
@iAmWillShepherd It doesn't seem like you addressed @niik's feedback here.
> The potential problem that we see with this is for the subset of our users that have a workflow similar to my own, i.e. they work primarily in one repository (rarely switch) and they keep their desktop instances running for a long time (weeks in my case) this approach will lead to us checking and incrementing the count pretty infrequently whereas the proposed solution for 1.7 we will check once every 24 hours regardless of whether they switch or not.

> The end result here could be that if the segment we're talking about isn't trivial in size (I would personally guess that it's a decent chunk of our user base) we'd see a spike in stashes created outside of desktop once we ship 1.7 which would be the opposite of what we'd want.